### DISCUSSION TITLE
Override nftables blocktype in jail.local?

### DISCUSSION THREAD

**Discussion by jhgarrison** [fail2ban/fail2ban#3698] [General]
Is it possible to override an nftables.conf [Init] parameter (blocktype=) in jail.local, or do I need to create action.d/nftables.local?

If so, what is the correct syntax?

**Comment by sebres** [comment]
Yes, all parameters can be supplied from jail.local.
See for instance how it is made with another parameters (like `port` or `protocol`): 
https://github.com/fail2ban/fail2ban/blob/0c125ec9c93d70f81a5fb6a18b84a380e1099333/config/jail.conf#L212

And you can do it for `action` or `banaction` (used in `action` or `action_` internally)... for instance:
```ini
[DEFAULT]
banaction = nftables[mode=multiport, blocktype="counter reject"]
banaction_allports = nftables[mode=allports, blocktype="counter reject"]

[some-jail]
action_ = %(known/action_)[blocktype="<known/blocktype> with whatever"]

[other-jail]
# single action (overwrite default):
action = %(known/action_)[blocktype="<known/blocktype> with whatever"]
# or single action (all parameters here):
action = nftables[mode=allports, port="%(port)s", protocol="%(protocol)s", chain="%(chain)s", blocktype="reject with whatever"]
```

However if you'd need it rather conditionally, e. g. depending on IP-family (like below), you'd need to do it in the local action:
```ini
[Init]
blocktype = reject with icmp type host-unreachable

[Init?family=inet6]
blocktype = reject with icmpv6 type no-route
```

You can also use `fail2ban-client -d | grep "$jail"` to check whether they are really overwritten.
